src/ImageSharp.Formats.Tiff/TiffDecoderCore.cs

```diff
@@ -0,0 +1,452 @@
+                {
+                    for (int i = 0 ; i < result.Length ; i++)
+                        result[i] = (int)ToInt16(bytes, i * TiffConstants.SizeOfShort);
+                    break;
+                }
+                case TiffType.SLong:
+                {
+                    for (int i = 0 ; i < result.Length ; i++)
+                        result[i] = ToInt32(bytes, i * TiffConstants.SizeOfLong);
+                    break;
+                }
+                default:
+                    throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a signed integer.");
+            }
+
+            return result;
+        }
+
+        public string ReadString(ref TiffIfdEntry entry)
+        {
+            if (entry.Type != TiffType.Ascii)
+                throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a string.");
+
+            byte[] bytes = ReadBytes(ref entry);
+            
+            if (bytes[entry.Count - 1] != 0)
+                throw new ImageFormatException("The retrieved string is not null terminated.");
+
+            return Encoding.UTF8.GetString(bytes, 0, (int)entry.Count - 1);
+        }
+
+        public Rational ReadUnsignedRational(ref TiffIfdEntry entry)
+        {
+            if (entry.Count != 1)
+                throw new ImageFormatException($"Cannot read a single value from an array of multiple items.");
+
+            return ReadUnsignedRationalArray(ref entry)[0];
+        }
+
+        public SignedRational ReadSignedRational(ref TiffIfdEntry entry)
+        {
+            if (entry.Count != 1)
+                throw new ImageFormatException($"Cannot read a single value from an array of multiple items.");
+
+            return ReadSignedRationalArray(ref entry)[0];
+        }
+
+        public Rational[] ReadUnsignedRationalArray(ref TiffIfdEntry entry)
+        {
+            if (entry.Type != TiffType.Rational)
+                throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a Rational.");
+
+            byte[] bytes = ReadBytes(ref entry);
+            Rational[] result = new Rational[entry.Count];
+
+            for (int i = 0 ; i < result.Length ; i++)
+            {
+                uint numerator = ToUInt32(bytes, i * TiffConstants.SizeOfRational);
+                uint denominator = ToUInt32(bytes, i * TiffConstants.SizeOfRational + TiffConstants.SizeOfLong);
+                result[i] = new Rational(numerator, denominator);
+            }
+
+            return result;
+        }
+
+        public SignedRational[] ReadSignedRationalArray(ref TiffIfdEntry entry)
+        {
+            if (entry.Type != TiffType.SRational)
+                throw new ImageFormatException($"A value of type '{entry.Type}' cannot be converted to a SignedRational.");
+
+            byte[] bytes = ReadBytes(ref entry);
+            SignedRational[] result = new SignedRational[entry.Count];
```

Comment:
This is definitely a place for Span-like stuff. Better encapsulation for data-chunk operations without unecessary allocations! I think I need to turn my 'BufferPointer<T>'  into 'BufferSpan<T>'. It will be almost System.Span compatible then!